crates/re_viewer/src/misc/caches/tensor_image_cache.rs

```diff
@@ -16,31 +16,63 @@ use crate::ui::{Annotations, DefaultColor, MISSING_ANNOTATIONS};
 
 // ---
 
-/// The `TensorImageView` is a wrapper on top of `re_log_types::Tensor`
+/// The `ColoredTensorView` is a wrapper on top of `re_log_types::Tensor`
 ///
 /// It consolidates the common operations of going from the raw tensor storage
 /// into an object that can be more natively displayed as an Image.
 ///
-/// The `dynamic_img` and `retained_img` are cached to keep the overhead low.
-///
 /// In the case of images that leverage a `ColorMapping` this includes conversion from
-/// the native Tensor type A -> Color32 which is stored for the cached dynamic /
-/// retained images.
-pub struct TensorImageView<'store, 'cache> {
+/// the native Tensor type A -> Color32.
+pub struct ColoredTensorView<'store, 'cache> {
+    /// Key used to retrieve this cached view
+    key: ImageCacheKey,
+
     /// Borrowed tensor from the data store
-    pub tensor: &'store dyn AsDynamicImage,
+    pub tensor: &'store Tensor,
 
     /// Annotations used to create the view
     pub annotations: &'store Arc<Annotations>,
 
-    /// DynamicImage helper for things like zoom
-    pub dynamic_img: Option<&'cache DynamicImage>,
+    /// Image with annotations applied and converted to Srgb
+    pub colored_image: Option<&'cache ColorImage>,
+
+    // For egui
+    // TODO(jleibs): This should go away. See [#506](https://github.com/rerun-io/rerun/issues/506)
+    pub retained_image: Option<&'cache RetainedImage>,
+}
 
-    /// For egui
-    pub retained_img: Option<&'cache RetainedImage>,
+impl<'store, 'cache> ColoredTensorView<'store, 'cache> {
+    pub fn texture_handle(&self, render_ctx: &mut RenderContext) -> Option<GpuTexture2DHandle> {
+        crate::profile_function!();
+        self.colored_image.map(|i| {
+            let mut s = nohash_hasher::NoHashHasher::<u64>::default();
+            self.key.hash(&mut s);
+            let texture_key = std::hash::Hasher::finish(&s);
```

Comment:
Create a `fn ImageCacheKey::hash64(&self) -> u64` that then is also used in `impl Hash for ImageCacheKey`

```suggestion
            let texture_key = self.key.hash64();
```